renderdoc/driver/vulkan/wrappers/vk_cmd_funcs.cpp

```diff
@@ -1152,27 +1152,32 @@ bool WrappedVulkan::Serialise_vkBeginCommandBuffer(SerialiserType &ser, VkComman
     // when loading, allocate a new resource ID for each push descriptor slot in this command buffer
     if(IsLoading(m_State))
     {
-      for(int p = 0; p < 2; p++)
+      BakedCmdBufferInfo::PushDescriptorBindPointArray &bpMap =
+          m_BakedCmdBufferInfo[BakedCommandBuffer].pushDescriptorID;
+      for(auto bp = bpMap.begin(); bp != bpMap.end(); ++bp)
       {
-        for(size_t i = 0; i < ARRAY_COUNT(BakedCmdBufferInfo::pushDescriptorID[p]); i++)
+        for(auto set = bp->begin(); set != bp->end(); ++set)
```

Comment:
I'd rather keep these as simple loops rather than switch to iterators. The outer loop could be changed from a hardcoded `2` to an ARRAY_COUNT over the base array.